tuist/tuist
# Fix default generated scheme arguments

### Short description 📝

The default scheme profile action doesn't inherit the run action's arguments in generated projects.

![profile-action-before](https://user-images.githubusercontent.com/11914919/101631705-b2e85d00-3a1c-11eb-85f0-09dda63e7b32.png)

### Solution 📦

Inspecting the code it looked like there were two related problems that caused this
- The default generated profile action was manually copying the run action arguments
    - This isn't needed as this is automatically handled by Xcode when the _"Use the Run action's arguments and environment"_ checkbox is selected
    - The default run action had an empty list of arguments as opposed to no arguments
    
We can address those by:
- No longer manually copying across the run action arguments
- No longer creating an empty list of arguments for the default generated run action incase no arguments or environment is specified

### Implementation 👩‍💻👨‍💻

- [x] Update `SchemeDescriptorGenerator` to no longer manually copy arguments
- [x] Update `AutogeneratedSchemesProjectMapper` to only create scheme arguments if values are set

Test Plan 🛠:

- Run `swift build && swift run tuist generate --path fixtures/ios_app_with_tests`
- Inspect the generated Project schemes
- Verify the profile action for App has the "Use the Run action's arguments and environment" checked
- Verify the raw `.xcscheme` doesn't contain any commandline arguments nor environment variable entries
- Tweak the App target to include some environment values (e.g. `environment: ["A": "B"]`)
- Re-generate the project
- Verify the App scheme in the generated project continues to specify the environment in the run action _(as it did before this change)_
